### PR TITLE
Con 2520 employer enter price

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Authorization.ModelBinding;
+using System;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Models.Apprentice
+{
+    public class EmployerLedChangeOfProviderRequest : IAuthorizationContextModel
```
Comment (hkf-tech) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs:8]: Not sure about the name of the class. But happy to discuss.

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        public long ApprenticeshipId { get; set; }
+        public long ProviderId { get; set; }
+        public string ProviderName { get; set; }
+
+        public int? NewStartMonth { get; set; }
+        public int? NewStartYear { get; set; }
+        public int? NewEndMonth { get; set; }
+        public int? NewEndYear { get; set; }
+        public int? NewPrice { get; set; }       
+        public bool? IsEdit { get; set; }
```
Comment (hkf-tech) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs:24]: Is this flag required

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        [FromRoute]
+        public string AccountHashedId { get; set; }
+        [FromRoute]
+        public string ApprenticeshipHashedId { get; set; }
+
+        public long ApprenticeshipId { get; set; }
+        public long ProviderId { get; set; }
+        public string ProviderName { get; set; }
+
+        public int? NewStartMonth { get; set; }
```
Comment (hkf-tech) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs:19]: should these be nullable?

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+﻿@using SFA.DAS.CommitmentsV2.Shared.Extensions
+@using  SFA.DAS.EmployerCommitmentsV2.Web.RouteValues
+@model SFA.DAS.EmployerCommitmentsV2.Web.Models.Apprentice.WhatIsTheNewPriceViewModel
+
+@{
+    ViewBag.Title = "What is the agreed price of completing the training with  new training provider name? ";
```
Comment (hkf-tech) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml:6]: "new training provider name" needs replacing with the actual training provider name

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                        For example, 5000
+                    </span>
+
+                    @Html.ValidationMessageFor(m => m.NewPrice, null, new { @class = "govuk-error-message", id = "error-message-" + Html.IdFor(m => m.NewPrice) })
+                    <input asp-for="NewPrice" class="govuk-input govuk-!-width-one-third @Html.AddClassIfPropertyInError(x=>x.NewPrice, "govuk-input--error input-validation-error")" />
+
+                </fieldset>
+            </div>
+
+            <button class="govuk-button" type="submit"> Continue </button>
```
Comment (hkf-tech) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml:46]: Id to keep testers happy

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                    <p class="govuk-body">
+                        This is the price you agreed with @Model.ProviderName to complete the remainder of the apprenticeship training and assessment.
+                    </p>
+
+                    <span class="govuk-hint">
+                        For example, 5000
+                    </span>
+
+                    @Html.ValidationMessageFor(m => m.NewPrice, null, new { @class = "govuk-error-message", id = "error-message-" + Html.IdFor(m => m.NewPrice) })
+                    <input asp-for="NewPrice" class="govuk-input govuk-!-width-one-third @Html.AddClassIfPropertyInError(x=>x.NewPrice, "govuk-input--error input-validation-error")" />
```
Comment (hkf-tech) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml:41]: Id to keep testers happy

**Review by hkf-tech — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs (hunk 1 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs:8]: CON-2518, CON-2519, CON-2520 - we have decided together to use this Model, can change it if required

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs (hunk 2 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs:24]: CON-2518, CON-2519, CON-2520 - we have decided together to use this Model, can change it if required

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml (hunk 4 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml]: done

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml (hunk 5 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml]: Good point :-), done

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml (hunk 6 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/WhatIsTheNewPrice.cshtml]: done

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs (hunk 1 above)
Comment (hkf-tech) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/EmployerLedChangeOfProviderRequest.cs:8]: that is fine, if it is done for reason.

**Review by hkf-tech — APPROVED** [review]

**Comment by Najamuddin-Muhammad** [comment]
This branch was merged into the feature branch after approval hence this PR is no longer required

**Pull request closed without merging**